File: app/back/src/main/java/net/seijishikin/jp/normalize/shuushi_doc/v05/constants/ShuushiDocV5ShikinKanridantaiUmuConstants.java
package net.seijishikin.jp.normalize.shuushi_doc.v05.constants;

/**
 * 政治資金収支報告書V5資金管理団体有無定数
 */
public final class ShuushiDocV5ShikinKanridantaiUmuConstants {

    /** 無 */
    public static final short NASHI = 0;
    /** 有 */
    public static final short ARI = 1;

    /** 無 */
    public static final String NASHI_TEXT = "無";
    /** 有 */
    public static final String ARI_TEXT = "有";

    private ShuushiDocV5ShikinKanridantaiUmuConstants() {

    }

    /**
     * 表示ラベルを取得する
     *
     * @param keyValue 設定値
     * @return 表示テキスト
     * @throws IllegalArgumentException このクラスで使用する定数でない場合
     */
    public static final String getLabel(final short keyValue) throws IllegalArgumentException { // NOPMD UnChecked

        switch (keyValue) {

            case NASHI:
                return NASHI_TEXT;

            case ARI:
                return ARI_TEXT;

            default:
                throw new IllegalArgumentException("Unexpected value: " + keyValue);
        }
    }

}

File: app/seijishikin-jp-normalize/src/components/main/constants/shuushi_doc/v05/shuushiDocV5ShikinKanridantaiUmuConstants.ts
export class ShuushiDocV5ShikinKanridantaiUmuConstants {

    /** 無 */
    static readonly NASHI = 0;
    /** 有 */
    static readonly ARI = 1;

    /** 無 */
    static readonly NASHI_TEXT = "無";
    /** 有 */
    static readonly ARI_TEXT = "有";

    /**
     * 値に該当するラベルを作成する
     * @param keyValue 値
     * @returns ラベル
     */
    public static getLabel(keyValue: number): string {

        switch (keyValue) {
            case ShuushiDocV5ShikinKanridantaiUmuConstants.NASHI:
                return ShuushiDocV5ShikinKanridantaiUmuConstants.NASHI_TEXT;

            case ShuushiDocV5ShikinKanridantaiUmuConstants.ARI:
                return ShuushiDocV5ShikinKanridantaiUmuConstants.ARI_TEXT;

            default:
                return "";
        }
    }
}
